src/mono/mono/component/debugger-engine.c

```diff
@@ -975,10 +975,11 @@ mono_de_ss_update (SingleStepReq *req, MonoJitInfo *ji, SeqPoint *sp, void *tls,
 	if (minfo)
 		loc = mono_debug_method_lookup_location (minfo, sp->il_offset);
 
-	if (!loc) {
-		PRINT_DEBUG_MSG (1, "[%p] No line number info for il offset %x, continuing single stepping.\n", (gpointer) (gsize) mono_native_thread_id_get (), sp->il_offset);
+	if (!loc) { //we should not continue single stepping because the client side can have symbols loaded dynamically
+		PRINT_DEBUG_MSG (1, "[%p] No line number info for il offset %x, don't know if it's in the same line single stepping.\n", (gpointer) (gsize) mono_native_thread_id_get (), sp->il_offset);
 		req->last_method = method;
-		hit = FALSE;
+		req->last_line = -1;
+		return hit;
```

Comment:
Before this change the step would not stop because runtime cannot check if it's in the same line (because runtime doesn't have debug information), so it would continue single stepping until finding this information, not pausing in a symbol that was only loaded by IDE.
Right now runtime doesn't have debug information, but sends the "pause" to the IDE and the IDE should decide if he wants to continue single stepping if it's in the same line.